include/pybind11/functional.h

```diff
@@ -75,6 +84,36 @@ struct type_caster<std::function<Return(Args...)>> {
             // PYPY segfaults here when passing builtin function like sum.
             // Raising an fail exception here works to prevent the segfault, but only on gcc.
             // See PR #1413 for full details
+        } else {
+            // Check number of arguments of Python function
+            auto getArgCount = [&](PyObject *obj) {
+                // This is faster then doing import inspect and inspect.signature(obj).parameters
+                auto *t = PyObject_GetAttrString(obj, "__code__");
```

Comment:
Won't this cause a memory leak as GetAttrString returns a new reference, but you never Py_DECREF?

Why not use pybind11's object wrappers to make sure you never leak memory? (https://pybind11.readthedocs.io/en/stable/reference.html#_CPPv46object)

Reply:
Yes you are right I used `PyObject_GetAttrString` and `.attr` where i don't have to check if the attribute.